pkg/api/componentreadiness/component_report.go

```diff
@@ -608,6 +677,180 @@ func (s *sampleQueryGenerator) queryTestStatus() (crtype.ReportTestStatus, []err
 	return crtype.ReportTestStatus{SampleStatus: sampleStatus}, errs
 }
 
+type fallbackTestQueryReleasesGenerator struct {
+	client                     *bqcachedclient.Client
+	cacheOption                cache.RequestOptions
+	commonQuery                string
+	groupByQuery               string
+	queryParameters            []bigquery.QueryParameter
+	allJobVariants             crtype.JobVariants
+	BaseRelease                string
+	BaseStart                  time.Time
+	BaseEnd                    time.Time
+	CachedFallbackTestStatuses crtype.FallbackReleases
+}
+
+func (f *fallbackTestQueryReleasesGenerator) getTestFallbackReleases() (*crtype.FallbackReleases, []error) {
+	wg := sync.WaitGroup{}
+	fallbackRelease := f.BaseRelease
+	f.CachedFallbackTestStatuses = newFallbackReleases()
+
+	releases, errs := GetReleaseDatesFromBigQuery(f.client)
+
+	if errs != nil {
+		return nil, errs
+	}
+
+	var err error
+	// currently gets current base plus previous 3
+	// current base is just for testing but use could be
+	// extended to no longer require the base query
+	for i := 0; i < 4; i++ {
+
+		// currently fetching the baseRelease via fallback to
+		// perform validation of the test data
+		// this gets uncommented when ready
+		// and the section at the bottom removed
+		// fallbackRelease, err = previousRelease(fallbackRelease)
+		// if err != nil {
+		// 	log.WithError(err).Errorf("Failure determining fallback release for %s", fallbackRelease)
+		// 	continue
+		// }
+
+		start := f.BaseStart
+		end := f.BaseEnd
+		var crRelease *crtype.Release
+
+		for _, release := range releases {
+			if release.Release == fallbackRelease {
+				crRelease = &release
+				if release.GADate != nil && release.GA30DayPrior != nil {
+					end = *release.GADate
+					start = *release.GA30DayPrior
+				}
```

Comment:
So unless if the fallback feature is enabled, regardless what base date range you choose, we may swap it out to the 30 days prior to the selected release's GA?

I'm wondering how we can smooth over the UX there. We've got the relative strings somewhat available via the views work (now-7d, ga-30d), but users can't specify that in the UI today. Seems like we *must* preserve ability to select specific date ranges, we can't just assume to let someone type/select now/ga-Xd because someone will eventually want a specific past date that is neither. Personally I would not be opposed to just typing in ga-30d in a textfield and adding support for 2024-10-14-30d (or some less dash ridden format) Frankly it's quicker than a date picker IMO.

Reply:
<img width="225" alt="image" src="https://github.com/user-attachments/assets/c9c31e08-0b38-4927-b2da-0509f96320b7">

I wish the basis release was a multi-select rather than having a fallback toggle of some kind.

For the dates, I'm not that convinced a text box is a great UI for someone who hasn't used CR before.  You could just disable the date picker when multi-release is selected and force GA. Date picker is more natural. Jira does dropdown filters by default, and has an advanced jql input field, which could also be an option.

BTW, we need to fix 30 days vs 4 weeks (28) and pick just one

Reply:
We only override dates for fallback queries.  We don't change the date for the selected basis.

We might consider something like `default`, or allow users to select specific releases / with dates.

I think much of this is likely separate from the initial effort though.  Lots we can do to standardize date selection / ui / etc.